scripts/illumina_run_parameter_parser.py

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import glob
+
+from argparse import ArgumentParser
+from datetime import datetime
+from genologics.lims import Lims
+from genologics.entities import Process
+from genologics.config import BASEURI, USERNAME, PASSWORD
+from flowcell_parser.classes import RunParser, RunParametersParser
+
+
+DESC = """EPP for parsing run paramters for Illumina MiSeq, NextSeq and NovaSeq runs"""
+
+
+def fetch_fc(process):
+    fc_id = ''
+    if process.parent_processes()[0].type.name == 'Load to Flowcell (NextSeq v1.0)':
+        if process.parent_processes()[0].udf['Experiment Name']:
+            fc_id = process.parent_processes()[0].udf['Experiment Name'].upper()
+        else:
+            sys.stderr.write("Experiment Name is empty in the associated Load to Flowcell (NextSeq v1.0) step.")
+            sys.exit(2)
+    elif process.parent_processes()[0].type.name == 'Denature, Dilute and Load Sample (MiSeq) 4.0':
+        if process.parent_processes()[0].udf['Flowcell ID']:
+            fc_id = process.parent_processes()[0].udf['Flowcell ID'].upper()
+        else:
+            sys.stderr.write("Flowcell ID is empty in the associated Denature, Dilute and Load Sample (MiSeq) 4.0 step.")
+            sys.exit(2)
+    elif process.parent_processes()[0].type.name == 'Load to Flowcell (NovaSeq 6000 v2.0)':
+        fc_id = process.parent_processes()[0].output_containers()[0].name
```

Comment:
Will the name be populated at all times? Just asking because of all the else conditions above.

Reply:
Yes, `output_containers` is a compulsory variable and it cannot be missed

Reply:
👍